controllers/apps/transformer_component_workload.go

```
@@ -644,19 +644,22 @@ func (r *componentWorkloadOps) leaveMember4ScaleIn() error {
 	}
 
 	var leaveErrors []error
+	var leaveDone []string
 	for _, pod := range podsToMemberLeave {
 		if podsToMemberjoin.Has(pod.Name) {
-			leaveErrors = append(leaveErrors, fmt.Errorf("pod %s is in memberjoin process", pod.Name))
+			r.reqCtx.Log.Info(fmt.Sprintf("pod %s is in memberjoin process", pod.Name))
```

The previous code here returns an error, which seems to be some kind of protection mechanism to prevent scale-in and scale-out from running simultaneously. Shouldn't the proper solution be to check why scale-out didn't clean up the `workloads.kubeblocks.io/memberjoin-in-processing` label?

@wangyelei please take a look.

---

The reason why scale-out did not successfully clear the `workloads.kubeblocks.io/memberjoin-in-processing` annotation was due to being blocked by the aforementioned error. The essence of this bug is that during a scale-in operation, there should be no need to report an error just because a node has not completed member join. Instead, it should simply ignore that node for this scale-in attempt and allow the reconcile process to complete the member join. Since member join is inherently a post-action of scale-out, the implementation in version 0.9 would definitely require two reconcile cycles to finish. If the scale-in operation fails due to this check, it would trap the system in a deadlock loop: scale-in check error -> ITS skips the update -> annotation cannot be cleared -> scale-in check error, and so on.